Documentation for operator privileges (#68964)
elastic/elasticsearch: x-pack/docs/en/security/operator-privileges/operator-only-functionality.asciidoc

[role="xpack"]
[testenv="enterprise"]
[[operator-only-functionality]]
=== Operator-only functionality

> There are two broad categories of operator-only functionalities: APIs and dynamic cluster Settings.

Reviewer: Are operator-only functionalities different from operator privileges? I see the terms are both used in the documentation, but seem the same to me. Would it be worth to clarify?

Author: The phrase "operator-only fuctionalities" here represents the things that are only actionable/updatable by an operator, i.e. the things protected by operator privileges. I struggled to come up with better terms. "operator-only resources"? "operator-only actions"? "operator-privileges protected stuff"? I am open to suggestions. Thanks!

Reviewer: So if I get it correctly, operator privileges is the feature name, while operator-only functionalities refer to list items. I'm not against the term "functionalities", I was just wondering if the term should have been "privileges" instead.

Author: I added some more words in the beginning of this section. Hopefully it makes things clear.

[role="xpack"]
[testenv="enterprise"]
[[operator-only-functionality]]
=== Operator-only functionality

> Operator-only functionalities are APIs or dynamic cluster settings protected by operator privileges.

Reviewer 2: I think "operator only functionality" is fine but functionalities (in plural) sounds a little weird to me. Maybe we could rephrase some of these sentences so that they are in the singular instead.

```suggestion
Operator privileges provides protection for APIs and dynamic cluster settings.
Any API or cluster setting that is protected by operator privileges is known as _operator-only functionality_
```

> When the operator privileges feature is enabled, they can only be actioned/updated by operators.

Reviewer 2: ```suggestion
When the operator privileges feature is enabled, operator-only APIs may only be executed by an
operator user. Likewise, operator-only settings may only be updated by operator users.
```

The list of operator-only APIs and dynamic cluster settings are pre-determined in the codebase.
They may evolve in future releases, but are otherwise fixed in a given {es} version.

[[operator-only-apis]]
==== Operator-only APIs
* <<voting-config-exclusions,Voting exclusions APIs>>
* <<delete-license,Delete license API>>
* <<update-license,Update license API>>
* <<autoscaling-apis,Autoscaling APIs>>

[[operator-only-dynamic-cluster-settings]]
==== Operator-only dynamic cluster settings

> * <<ip-filtering,IP filtering>>

Reviewer 2: ```suggestion
* All <<ip-filtering,IP filtering>> settings
```

> * Many of the dynamic <<ml-settings, Machine learning settings>>:

Reviewer 2: ```suggestion
* The following the dynamic <<ml-settings, Machine learning settings>>:
```

* <<voting-config-exclusions,Voting exclusions APIs>>
* <<delete-license,Delete license API>>
* <<update-license,Update license API>>
* <<autoscaling-put-autoscaling-policy,Put autoscaling policy API>>

> * <<autoscaling-delete-autoscaling-policy,Delete autoscaling policy>>

Reviewer 2: This is missing "API"
```suggestion
* <<autoscaling-delete-autoscaling-policy,Delete autoscaling policy API>>
```
But won't you get the right outcome just by relying on asciidoctor's auto link text?
```suggestion
* <<autoscaling-delete-autoscaling-policy>>
```

Author: Thanks that will do. I copy/pasted from https://raw.githubusercontent.com/elastic/elasticsearch/7.12/docs/reference/autoscaling/apis/autoscaling-apis.asciidoc
